message_ix_models/model/water/data/water_for_ppl.py

```diff
@@ -171,6 +177,91 @@ def hist_cap(x: pd.Series, context: "Context", hold_cost: pd.DataFrame) -> list:
     ]
 
 
+def relax_growth_constraint(ref_hist, scen, cooling_df, g_lo, constraint_type):
+    """
+    Function to check if the parent technologies are shut down and require
+    relaxing the growth constraint.
+
+    Parameters:
+    ref_hist (pd.DataFrame): Historical data.
+    scen (Scenario): Scenario object to retrieve parameters.
+    cooling_df (pd.DataFrame): DataFrame containing cooling technologies and
+    their parent technologies.
+    g_lo (pd.DataFrame): DataFrame containing growth constraints.
+    constraint_type (str): Type of constraint to check ("activity" or "capacity").
+
+    Returns:
+    pd.DataFrame: Updated g_lo DataFrame with relaxed growth constraints.
+    """
+    if constraint_type == "activity":
+        year_type = "year_act"
+        bound_param = "bound_activity_up"
+    elif constraint_type == "new_capacity":
+        year_type = "year_vtg"
+        bound_param = "bound_new_capacity_up"
```

Comment:
```suggestion
    year_type = "year_act" if constraint_type == "activity" else "year_vtg"
    bound_param = "bound_activity_up" if constraint_type == "activity" else "bound_new_capacity_up" 
```
With proper type hinting above, it should be possible to shorten these conditionals. Please run `ruff`, though, and make sure the `constaint_type` is type-hinted correctly and consistently with the docstring of the function (`"capacity"` vs `"new_capacity"`).